### PR TITLE
Update amqplib definitions to v0.5.0

### PR DESCRIPTION
**Pull request by nfantone**
Please fill in this template.

- [x] Prefer to make your PR against the `types-2.0` branch.
- [ ] The package does not provide its own types, and you can not add them.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.
- [ ] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header.

If changing an existing definition:
- [x] Provide a URL to  documentation or source code which provides context for the suggested changes: https://github.com/squaremo/amqp.node/commit/e5c19d4da864954ba907daa00f0dd3910f239715
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

**Comment by nfantone** [comment]
Travis fails with an error unrelated to this PR:

```sh
For file:/home/travis/build/DefinitelyTyped/DefinitelyTyped/chocolatechipjs/tsconfig.json
\home\travis\build\DefinitelyTyped\DefinitelyTyped\chocolatechipjs\index.d.ts
	 Line 1363: All declarations of 'responseURL' must have identical modifiers.
```

**Comment by vvakame** [comment]
It is out problem. types-2.0/HEAD is back to green.

File: amqplib/index.d.ts [hunk 1]
```diff
@@ -1,13 +1,77 @@
...
-/// <reference types="when" />
 /// <reference types="node" />
 
-declare module "amqplib/properties" {
-    namespace Replies {
+declare module 'amqplib' {
+    import * as Promise from 'bluebird';
+    import * as events from 'events';
+    import shared = require('amqplib/properties');
+    import callback_api = require('amqplib/callback_api');
```
Comment (mnahkies) [inline amqplib/index.d.ts]: Think this callback_api import is unused?

**Comment by mnahkies** [comment]
Assuming passes CI after rebase / merge with upstream then :+1:

File: amqplib/index.d.ts (hunk 1 above)
Comment (nfantone) [inline amqplib/index.d.ts]: Completely right. Removed.

File: amqplib/index.d.ts [hunk 2]
```diff
@@ -1,13 +1,77 @@
...
+
+    export import Replies = shared.Replies;
```
Comment (mnahkies) [inline amqplib/index.d.ts]: My mistake, I thought that exporting in the fashion I suggested also imported them into the local scope

**Review by mnahkies — APPROVED** [review]

**Comment by nfantone** [comment]
@mnahkies Ok, it may be a little too late now - but I just realized that for some awkward reason, `Bluebird.Promise` is _not_ compatible with `Promise<T>` interface exported by `lib.es6.d.ts` - which is very inconvenient when mixing different libraries that expect or use `Promise`s (like RxJS).

When updating to `0.5.0`, I keep getting compiler errors like:

```sh
[ts] Argument of type 'Bluebird<void>' is not assignable to parameter of type 'Promise<void>'.
```

Thoughts?

**Pull request merged**